osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -344,21 +344,21 @@ public double ComputeFinalLegacyScore(ScoringMode mode, ScoreInfo scoreInfo, int
         /// <param name="bonusScore">The total bonus score.</param>
         /// <param name="totalBasicHitObjects">The total number of basic (non-tick and non-bonus) hitobjects in the beatmap.</param>
         /// <returns>The total score computed from the given scoring component ratios.</returns>
-        public double ComputeScore(ScoringMode mode, double accuracyRatio, double comboRatio, double bonusScore, int totalBasicHitObjects)
+        public int ComputeScore(ScoringMode mode, double accuracyRatio, double comboRatio, double bonusScore, int totalBasicHitObjects)
         {
             switch (mode)
             {
                 default:
                 case ScoringMode.Standardised:
                     double accuracyScore = accuracyPortion * accuracyRatio;
                     double comboScore = comboPortion * comboRatio;
-                    return (max_score * (accuracyScore + comboScore) + bonusScore) * scoreMultiplier;
+                    return (int)Math.Round((max_score * (accuracyScore + comboScore) + bonusScore) * scoreMultiplier);
```

Comment:
This seems like a bad idea, especially with classic scoring. On some maps on stable, ScoreV2 already has to be forced in order to avoid an integer overflow if the score gets too big. Let's not make the same mistake twice, shall we?

Should probably be a `long`, at least. And even then it needs checking how marathon maps behave with `long`.